### PR TITLE
Remove Join Condition Push CBO Rule since it was causing infinite cos…

### PR DESCRIPTION
**Pull request by jdye64**
…t based optimization for certain queries

### CODE REVIEW

**Comment by charlesbluca** [comment]
cc @randerzander

**Review by charlesbluca — APPROVED** [review]
Approving now but will hold off on merging until we get other confirmation that this resolves the infinite CBO

**Comment by randerzander** [comment]
This is resolved! Thanks all!

**Pull request merged**
